### ISSUE TITLE
Version from original file instead of other version

### ISSUE THREAD

**Issue by tomek-railwaymen** [carrierwaveuploader/carrierwave#1602]
I have the following code in my uploader:

``` ruby
  version :large do
    process :convert_to_png

    def full_filename(for_file = model.pdf.file)
      "large-#{File.basename(for_file, '.pdf')}.png"
    end

    def convert_to_png
      manipulate! do |img|
        img.format 'png' do |cmd|
          cmd.merge! ['-density', '200', '-quality', '100']
        end
        img = yield(img) if block_given?
        img
      end
    end
  end

  version :thumb, from_version: :large do
    process resize_to_fit: [200, 200]

    def full_filename(for_file = model.pdf.file)
      "thumb-#{File.basename(for_file, '.pdf')}.png"
    end
  end
```

Unfortunately `thumb` version is created based on original file (always in PDF format). That file is broken because it is PDF as well (instead of PNG, like `large` version). 

I noticed that `large` version is generated by command like `convert /tmp/mini_magick20150318-4571-1ma89u5.pdf[0] -density 200 -quality 100 /tmp/mini_magick20150318-4571-ifujor.png` and `thumb` uses PDF file: `mogrify -resize 200x200 /tmp/mini_magick20150318-4571-5or0in.pdf`

**Comment by cipater** [comment]
+1 `from_version` doesn't appear to be working. Which is to say that all versions look to be built off the original rather than version specified by `from_version`.

Given
```ruby
  process resize_to_limit: [1600, 1600]
  process :store_geometry

  version :optimized do
    process :pngquant
    process :optimize
    process :store_geometry
  end

  version :large, from_version: :optimized do
    process resize_to_limit: [1400, 1400]
    process :store_geometry
  end
```

The original file is 2.5MB. The optimized version is 900KB. And the large version is 1.8MB.

If the large version were in fact built from the optimized version, I would expect it to be smaller than 900KB, not twice the size.

**Comment by cianoswift** [comment]
Any updates on this? I'm still having this issue on v1.1.0

**Comment by mshibuya** [comment]
Fixed by #1879.
